lisa/src/test/java/it/unive/lisa/test/imp/tutorial/ConstantPropagation.java

```diff
@@ -0,0 +1,224 @@
+package it.unive.lisa.test.imp.tutorial;
+
+import it.unive.lisa.analysis.SemanticDomain.Satisfiability;
+import it.unive.lisa.analysis.SemanticException;
+import it.unive.lisa.analysis.nonrelational.BaseNonRelationalValueDomain;
+import it.unive.lisa.cfg.type.Type;
+import it.unive.lisa.symbolic.value.BinaryOperator;
+import it.unive.lisa.symbolic.value.Constant;
+import it.unive.lisa.symbolic.value.Identifier;
+import it.unive.lisa.symbolic.value.TernaryOperator;
+import it.unive.lisa.symbolic.value.UnaryOperator;
+
+public class ConstantPropagation extends BaseNonRelationalValueDomain<ConstantPropagation> {
+
+	public static final ConstantPropagation TOP = new ConstantPropagation(true, false);
+
+	private static final ConstantPropagation BOTTOM = new ConstantPropagation(false, true);
+
+	private final boolean isTop, isBottom;
+
+	private final Integer value;
```

Comment:
If it tracks only integers, it might be better calling this class `IntegerConstantPropagation`. Can we extend it to a `NumericalConstantPropagation` catching also other types of numeric values?